[PostgreSQL] Migrate PostgreSQL module to PostgreSQL flexible server GA API

DaeunYim opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

isra-fel commented on src/PostgreSql/readme.md:
```diff
@@ -165,7 +167,7 @@ directive:
         - DefaultValue
   - where:
-      model-name: ConfigurationAutoGenerated
+      model-name: ConffigurationAutoGenerated
```
I believe this line is a typo

DaeunYim replied:
Thanks fixed

isra-fel commented on src/PostgreSql/custom/Get-AzPostgreSqlFlexibleServerConnectionString.ps1:
```diff
@@ -99,7 +99,7 @@ function Get-AzPostgreSqlFlexibleServerConnectionString {
     process {
         $null = $PSBoundParameters.Remove('Client')
-        $postgreSql = Az.PostgreSql\Get-AzPostgreSqlFlexibleServer @PSBoundParameters
+        $postgreSql = aAz.PostgreSql\Get-AzPostgreSqlFlexibleServer @PSBoundParameters
```
There is a typo

DaeunYim replied:
Same here

isra-fel commented:
Also could you share some background of your PRs?

DaeunYim commented:
@isra-fel We are GAing PostgreSQL flexible server module. We have updated GA Swagger API and added some more features. Now PowerShell needs to be updated to based on that. We also want to include these to CloudShell once the product officially hit GA
The updated API is here
https://github.com/Azure/azure-rest-api-specs/tree/master/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2021-06-01

isra-fel commented:
Hey I don't think the previous comments were addressed - did you forget to push the commit?
Also same as MySql PR, you need to record test cases and check in updated documents.
Thanks

DaeunYim commented:
Looks this breaking change failure is due to GA migration.

 "Az.PostgreSql","Get-AzPostgreSqlFlexibleServer","Get-AzPostgreSqlFlexibleServer","0","1020","The cmdlet 'Get-AzPostgreSqlFlexibleServer' no longer has output type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'.","Make cmdlet 'Get-AzPostgreSqlFlexibleServer' return type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'."
  "Az.PostgreSql","Get-AzPostgreSqlFlexibleServerConnectionString","Get-AzPostgreSqlFlexibleServerConnectionString","0","2020","The cmdlet 'Get-AzPostgreSqlFlexibleServerConnectionString' no longer supports the type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated' for parameter 'InputObject'.","Change the type for parameter 'InputObject' back to 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'."
  "Az.PostgreSql","New-AzPostgreSqlFlexibleServer","New-AzPostgreSqlFlexibleServer","0","1020","The cmdlet 'New-AzPostgreSqlFlexibleServer' no longer has output type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'.","Make cmdlet 'New-AzPostgreSqlFlexibleServer' return type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'."
  "Az.PostgreSql","Restore-AzPostgreSqlFlexibleServer","Restore-AzPostgreSqlFlexibleServer","0","1020","The cmdlet 'Restore-AzPostgreSqlFlexibleServer' no longer has output type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'.","Make cmdlet 'Restore-AzPostgreSqlFlexibleServer' return type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'."
  "Az.PostgreSql","Update-AzPostgreSqlFlexibleServer","Update-AzPostgreSqlFlexibleServer","0","1020","The cmdlet 'Update-AzPostgreSqlFlexibleServer' no longer has output type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'.","Make cmdlet 'Update-AzPostgreSqlFlexibleServer' return type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Models.Api20200214Preview.IServerAutoGenerated'."
  "Az.PostgreSql","Update-AzPostgreSqlFlexibleServer","Update-AzPostgreSqlFlexibleServer","0","2000","The cmdlet 'Update-AzPostgreSqlFlexibleServer' no longer supports the parameter 'ReplicationRole' and no alias was found for the original parameter name.","Add the parameter 'ReplicationRole' back to the cmdlet 'Update-AzPostgreSqlFlexibleServer', or add an alias to the original parameter name."
  "Az.PostgreSql","Update-AzPostgreSqlFlexibleServer","Update-AzPostgreSqlFlexibleServer","0","2020","The cmdlet 'Update-AzPostgreSqlFlexibleServer' no longer supports the type 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Support.HaEnabledEnum' for parameter 'HaEnabled'.","Change the type for parameter 'HaEnabled' back to 'Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Support.HaEnabledEnum'."
  "Az.PostgreSql","Update-AzPostgreSqlFlexibleServer","Update-AzPostgreSqlFlexibleServer","0","2000","The cmdlet 'Update-AzPostgreSqlFlexibleServer' no longer supports the parameter 'SslEnforcement' and no alias was found for the original parameter name.","Add the parameter 'SslEnforcement' back to the cmdlet 'Update-AzPostgreSqlFlexibleServer', or add an alias to the original parameter name."
  "Az.PostgreSql","Update-AzPostgreSqlFlexibleServer","Update-AzPostgreSqlFlexibleServer","0","2000","The cmdlet 'Update-AzPostgreSqlFlexibleServer' no longer supports the parameter 'StorageAutogrow' and no alias was found for the original parameter name.","Add the parameter 'StorageAutogrow' back to the cmdlet 'Update-AzPostgreSqlFlexibleServer', or add an alias to the original parameter name."
  "Az.PostgreSql","Update-AzPostgreSqlFlexibleServer","Update-AzPostgreSqlFlexibleServer","0","1050","The parameter set 'UpdateExpanded' for cmdlet 'Update-AzPostgreSqlFlexibleServer' has been removed.","Add parameter set 'UpdateExpanded' back to cmdlet 'Update-AzPostgreSqlFlexibleServer'."
  "Az.PostgreSql","Update-AzPostgreSqlFlexibleServer","Update-AzPostgreSqlFlexibleServer","0","1050","The parameter set 'UpdateViaIdentityExpanded' for cmdlet 'Update-AzPostgreSqlFlexibleServer' has been removed.","Add parameter set 'UpdateViaIdentityExpanded' back to cmdlet 'Update-AzPostgreSqlFlexibleServer'."
  "Az.PostgreSql","Update-AzPostgreSqlFlexibleServer","Update-AzPostgreSqlFlexibleServer","0","1050","The parameter set '__AllParameterSets' for cmdlet 'Update-AzPostgreSqlFlexibleServer' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'Update-AzPostgreSqlFlexibleServer'."

How can I resolve this?

isra-fel commented:
/azp run azure-powershell - powershell-core

isra-fel reviewed (approved):
LGTM

isra-fel commented:
/azp run /azp run azure-powershell - powershell-core

isra-fel commented:
/azp run azure-powershell - powershell-core

isra-fel commented:
/azp run azure-powershell - powershell-core

DaeunYim commented:
/azp run /azp run azure-powershell - powershell-core
